### ISSUE TITLE
The blog plugin doesn't respect the `hide` property of an `index.md` page when pagination is in effect.

### ISSUE THREAD

**Issue by hellt** [squidfunk/mkdocs-material#4396]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

The blog plugin doesn't respect `hide` property of an index.md page when using pagination.

### Expected behaviour

Elements that are marked to be hidden in `index.md` continue to be hidden on paginated views (`/page/X`).

### Actual behaviour

`navigation` and `toc` elements are not following the parameters set in `index.md`

### Steps to reproduce

1. add more posts to trigger paginated view
2. see the result

### Package versions

- Material: 8.5.3-insiders-4.23.6

### Configuration

```yaml
blog pluging enabled.

index.md contents


---
hide:
  - navigation

---
# Blog

```
```


### System information

- Operating system: Mac OS
- Browser: Chrome

**Comment by squidfunk** [comment]
Thanks for reporting. I agree that the paginated pages should inherit the metadata from the blog index page. Fixed in squidfunk/mkdocs-material-insiders@a9af801412007f99f384ced687dc09c1bfeb4b69. Additionally, there's a new setting called `pagination_keep_content`, that specifies whether the paginated pages should also include the content of the `blog/index.md`, defaulting to `false`:

``` yaml
plugins:
  - blog:
      pagination_keep_content: true
```

**Comment by squidfunk** [comment]
The fix was released as part of 8.5.3+insiders.4.24.0.
